src/OrchardCore.Modules/OrchardCore.PublishLater/Migrations.cs

```diff
@@ -1,59 +1,84 @@
+            .Column<DateTime>(nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
+        );
+
+        SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
+            .CreateIndex($"IDX_{nameof(PublishLaterPartIndex)}_{nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc)}",
+                "DocumentId",
+                nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
+        );
+
+        // Shortcut other migration steps on new content definition schemas.
+        return 2;
+    }
 
-            SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
-                .AddColumn<DateTime>(nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
-            );
+    // This code can be removed in a later version.
+    public int UpdateFrom1()
+    {
+        // The 'ScheduledPublishUtc' column and related index are kept on existing databases,
+        // this because dropping an index and altering a column don't work on all providers.
+
+        SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
+            .AddColumn<DateTime>(nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
+        );
+
+        SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
+            .CreateIndex($"IDX_{nameof(PublishLaterPartIndex)}_{nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc)}",
+                "DocumentId",
+                nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
+        );
+
+        return 2;
+    }
+
+    public int UpdateFrom2()
+    {
+        SchemaBuilder.CreateMapIndexTable<PublishLaterPartIndex>(table => table
+            .Column<string>(nameof(PublishLaterPartIndex.ContentItemId))
+            .Column<bool>(nameof(PublishLaterPartIndex.Latest))
+            .Column<bool>(nameof(PublishLaterPartIndex.Published))
+            .Column<DateTime>(nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
+        );
+
+        SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table =>
+        {
+            table.DropIndex($"IDX_{nameof(PublishLaterPartIndex)}_{nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc)}");
 
-            SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
-                .CreateIndex($"IDX_{nameof(PublishLaterPartIndex)}_{nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc)}",
-                    "DocumentId",
-                    nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
-            );
+            table.CreateIndex($"IDX_{nameof(PublishLaterPartIndex)}_{nameof(ContentItemIndex.DocumentId)}",
+                "Id",
+                nameof(ContentItemIndex.DocumentId),
+                nameof(PublishLaterPartIndex.ContentItemId),
+                nameof(PublishLaterPartIndex.Latest),
+                nameof(PublishLaterPartIndex.Published),
+                nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc));
+        });
```

Comment:
Here you should first drop the existing non clustured IDX_ index, something like this.

        SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
            .DropIndex("IDX_...")
        );

Then rebuild it with all the inculded columns, Published/Latest at the end.

Note: In one project I needed to drop the non clustered IDX_ at the very beginning, but I think it was needed before removing columns (included in this IDX_) of the related Index table, which is not the case here.

Anyway would need to be tried, first with a fresh setup without the changes in this PR, then restart the app to see what happens when this migration step is executed. Then finally re-do a fresh setup with all the updates to see if the new Create method is okay. At least with Sqlite and Sql Server.

Reply:
Again you are right, it's my fingers fault ;)

Reply:
No problem ;)

Will review again when you will update it, but take your time, no pressure.

Reply:
I will update them now, I already switch to branch but waiting for VS to load the projects ;)

Reply:
@jtkech if we drop the index, then we can create the columns from ground up, am I right instead of adding columns using `AddColumn()`

Reply:
Yes, you have to use `.AddColumn()`, then also `.DropIndex()` to first remove the inner non clustered index, see my other comments for more details, e.g. https://github.com/OrchardCMS/OrchardCore/pull/10935#issuecomment-1221855101, then finally `.CreateIndex("IDX_Something)` to rebuild the inner index by including all columns.